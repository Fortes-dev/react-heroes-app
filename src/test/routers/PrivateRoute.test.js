import { mount } from "enzyme";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";
import { PrivateRoute } from "../../routers/PrivateRoute";

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    Navigate: () => <h1>Navigate</h1>
}));


describe('Suite en PrivateRoute', () => {
    
    Storage.prototype.setItem = jest.fn();

    test('Debe mostrar el componente si está autenticado y guardar en el localstorage', () => {
        
        const contextValue = {
            user: {
                logged: true,
                name: 'Juanito'
            },

        }
        
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={['/']}>
                    <PrivateRoute>
                        <h1>PrivateComponent</h1>
                    </PrivateRoute>
                </MemoryRouter>
            </AuthContext.Provider>
        );
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text().trim()).toBe('PrivateComponent');
        expect(Storage.prototype.setItem).toHaveBeenCalledWith('lastPath', '/');
        
        

    });

    test('Debe de bloquear el componente si no está autenticado', () => {
        
        const contextValue = {
            user: {
                logged: false,
            },

        }
        
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={['/']}>
                    <PrivateRoute>
                        <h1>PrivateComponent</h1>
                    </PrivateRoute>
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper.find('h1').text().trim()).toBe('Navigate');

    });

});