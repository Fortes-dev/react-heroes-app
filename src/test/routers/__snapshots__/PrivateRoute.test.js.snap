// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suite en PrivateRoute Debe mostrar el componente si est√° autenticado y guardar en el localstorage 1`] = `
<h1>
  PrivateComponent
</h1>
`;
